apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-commons/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/commons/interceptor/AbstractMethodInterceptor.java

```diff
@@ -198,56 +198,58 @@ public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allA
+                        statusCode = ((HttpServletResponse) response).getStatus();
+                    } else if (ServerHttpResponse.class.isAssignableFrom(response.getClass())) {
+                        if (IS_SERVLET_GET_STATUS_METHOD_EXIST) {
+                            statusCode = ((ServerHttpResponse) response).getRawStatusCode();
+                        }
+                        Object context = runtimeContext.get(REACTIVE_ASYNC_SPAN_IN_RUNTIME_CONTEXT);
+                        if (context != null) {
+                            ((AbstractSpan[]) context)[0] = span.prepareForAsync();
+                        }
                     }
-                }
 
-                if (statusCode != null && statusCode >= 400) {
-                    span.errorOccurred();
-                    Tags.HTTP_RESPONSE_STATUS_CODE.set(span, statusCode);
-                }
+                    if (statusCode != null && statusCode >= 400) {
+                        span.errorOccurred();
+                        Tags.HTTP_RESPONSE_STATUS_CODE.set(span, statusCode);
+                    }
 
-                runtimeContext.remove(REACTIVE_ASYNC_SPAN_IN_RUNTIME_CONTEXT);
-                runtimeContext.remove(REQUEST_KEY_IN_RUNTIME_CONTEXT);
-                runtimeContext.remove(RESPONSE_KEY_IN_RUNTIME_CONTEXT);
-                runtimeContext.remove(CONTROLLER_METHOD_STACK_DEPTH);
-            }
+                    runtimeContext.remove(REACTIVE_ASYNC_SPAN_IN_RUNTIME_CONTEXT);
+                    runtimeContext.remove(REQUEST_KEY_IN_RUNTIME_CONTEXT);
+                    runtimeContext.remove(RESPONSE_KEY_IN_RUNTIME_CONTEXT);
+                    runtimeContext.remove(CONTROLLER_METHOD_STACK_DEPTH);
+                }
 
-            // Active HTTP parameter collection automatically in the profiling context.
-            if (!SpringMVCPluginConfig.Plugin.SpringMVC.COLLECT_HTTP_PARAMS && span.isProfiling()) {
-                if (HttpServletRequest.class.isAssignableFrom(request.getClass())) {
-                    RequestUtil.collectHttpParam((HttpServletRequest) request, span);
-                } else if (ServerHttpRequest.class.isAssignableFrom(request.getClass())) {
-                    RequestUtil.collectHttpParam((ServerHttpRequest) request, span);
+                // Active HTTP parameter collection automatically in the profiling context.
+                if (!SpringMVCPluginConfig.Plugin.SpringMVC.COLLECT_HTTP_PARAMS && span.isProfiling()) {
+                    if (HttpServletRequest.class.isAssignableFrom(request.getClass())) {
+                        RequestUtil.collectHttpParam((HttpServletRequest) request, span);
+                    } else if (ServerHttpRequest.class.isAssignableFrom(request.getClass())) {
+                        RequestUtil.collectHttpParam((ServerHttpRequest) request, span);
+                    }
                 }
+            } finally {
+                ContextManager.stopSpan();
```

Comment:
Are you only adding this `try/finally` to stop span? Could you share when this last step goes missed?

Reply:
> Are you only adding this `try/finally` to stop span? Could you share when this last step goes missed?

Yes, I only did this. 

The last step goes missed when the response == null and the exception ServletResponseNotFoundException throws.

It is complicated to reproduced why the response becomes null, since it is only appear in a few requests. And if I debug into the request, drop the frame, the response becomes not null.

Maybe I could read more springMVC source codes later and find out the root cause.

Reply:
OK, to add this to keep tracing context safe is correct. I just think, this may not fix the real issue about why response is null, and we can't get all response related information.